async_js.md

```diff
@@ -0,0 +1,289 @@
+
+```js
+async function myFunc() {
+  return "biror qiymat";
+}
+
+myFunc().then((qiymat) => console.log(qiymat));
+```
+
+Yuqoridagi misolda biz async kalit so'zidan foydalandiku lekin await kalit so'zidan foydalanmadik. Aslida async har doim await kalit so'zi bilan birga keladi. Ya'ni async orqali funksiya yaratayapmizmi uni ichida await'dan ham foydalanishimiz kerak.
+
+`await` - ham o'zi bir operator hisoblanadi. Bu operator bizga promise'ning resolved qiymatini qaytarib beradi.
+
+```js
+async function myFunc() {
+  const qiymat = await "biror qiymat";
+  return qiymat;
+}
+
+myFunc().then((qiymat) => console.log(qiymat));
+```
+
+Bundan tashqari await operatori promise holati pending holatdan boshqa holatga o'tgangacha jarayonni to'xtatib turish vazifasini bajaradi. Shuning uchun ham async funksiyalarda await kalit so'zidan foydalanishni esdan chiqmaslik muhimdir.
+
+```js
+const meningPromise = () => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve("Muvoffiqiyatli qiymat");
+    }, 1000);
+  });
+};
+
+async function awaitBilan() {
+  const natija = await meningPromise();
+  console.log(natija);
+}
+
+async function awaitSiz() {
+  const natija = meningPromise();
+  console.log(natija);
+}
+
+awaitBilan();
+awaitSiz();
+```
+
+Quyidagi na'munaviy misolda siz async \ await sintaksisini qulayligini ko'rishingiz mumkin.
+
+```js
+function odatiySintaks() {
+  birinchiPromise()
+    .then((birinchiQiymat) => {
+      ...
+      return ikkinchiPromise(birinchiQiymat);
+    })
+    .then((ikkinchiQiymat) => {
+      ...
+      return uchinchiPromise(ikkinchiQiymat);
+    })
+    .then((uchinchiQiymat) => {
+      ...
+    });
+}
+```
+
+```js
+async function maxsusHolat() {
+  const birinchiQiymat = await birinchiPromise();
+  ...
+  const ikkinchiQiymat = await ikkinchiPromise(birinchiQiymat);
+  ...
+  const uchinchiQiymat = await uchinchiPromise(ikkinchiQiymat);
+}
+```
+
+Ko'rib turganizdek 2 - usulda kod yozish ancha qulaylik yaratadi.
+
+async funksiyasi ham oddiy bir promise qaytarib beruvchi funksiya ekanligini hisobga olsak, biz promise rad etilish sabablarini o'sha odatiy `catch()` metodi orqali tutsak ham bo'ladi.
+
+```js
+async function xatoniTutish() {
+  const qiymat = await birorPromise();
+  ...
+  return qiymat;
+}
+
+const ozgaruvchi = xatoniTutish();
+ozgaruvchi.catch((xato) => console.log(xato));
+```
+
+Bundan tashqari promise'ning rad etilish sababini `try...catch` sintaksisi orqali ham tutsak bo'ladi.
+
+```js
+async function xatoniTutish() {
+  try {
+    const qiymat = await birorPromise();
+    ...
+  } catch (xato) {
+    console.log(xato);
+  }
+}
+```
+
+Yuqoridagi ikkala yo'l ham xatoni tutish uchun to'g'ri yo'l hisoblanadi. Lekin `async...await` sintaksisi bilan `try...catch` sintaksisi ishlatilishi keng dasturchilar orasida keng tarqalgan.
```

Comment:
oxirida xulosani ham qo'shib qo'ysangiz yaxshi bo'lardi, aynan ushbu maqolada nima haqida gap ketganligini

Reply:
Aha, xulosa ham qo'shamiz

Reply:
Umuman olganda post  yaxshi yozilgan , shu men yozgan commentlarga bir qarab ko'ring , o'zizga yoqqanini to'g'irlab yozaverasiz